### ISSUE TITLE
Cannot filter admin user by ID

### ISSUE THREAD

**Issue by Jeroenll** [magento/magento2#23266]
### Preconditions (*)
1. Magento v2.3.1

### Steps to reproduce (*)
1. Filter admin user grid (System > All Users) by ID.

### Expected result (*)
1. Filtered user is shown.

### Actual result (*)
1. 500 error is logged

```
[2019-06-14 13:00:29] main.CRITICAL: SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'user_id' in where clause is ambiguous, query was: SELECT COUNT(*) FROM `admin_user` AS `main_table`
 LEFT JOIN `authorization_role` AS `user_role` ON main_table.user_id = user_role.user_id AND user_role.parent_id != 0
 LEFT JOIN `authorization_role` AS `detail_role` ON user_role.parent_id = detail_role.role_id WHERE (`user_id` LIKE '%106%') [] []
```

**Comment by Jeroenll** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @JeroenVanLeusden. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @JeroenVanLeusden, here is your [Magento instance](https://i-23266-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23266-2-3-develop.instances.magento-community.engineering/admin](https://i-23266-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
Not reproduced on 2.2-develop.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-qa-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17506` were created

**Issue Available**: @engcom-qa-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @JeroenVanLeusden. Thank you for your report. 
The issue has been fixed in magento/magento2#23267 by @JeroenVanLeusden in 2.3-develop branch
Related commit(s):
 - [6e497a6b13fad5b01dcc9f33da6537f906b33537](https://github.com/magento/magento2/commit/6e497a6b13fad5b01dcc9f33da6537f906b33537)

The fix will be available with the upcoming 2.3.3 release.
